time_tracker_api/time_entries/time_entries_repository.py

```diff
@@ -247,35 +240,39 @@ def find_interception_with_date_range(
             "owner_id": owner_id,
             "tenant_id": tenant_id,
         }
-        params = [
-            {"name": "@start_date", "value": start_date},
-            {"name": "@end_date", "value": end_date or current_datetime_str()},
-            {"name": "@ignore_id", "value": ignore_id},
+        end_date = end_date or current_datetime_str()
+
+        not_equal_conditions = [
+            {
+                "field_name": "start_date",
+                "compare_field_name": "end_date",
+                "compare_field_value": end_date,
+            },
+            {
+                "field_name": "end_date",
+                "compare_field_name": "start_date",
+                "compare_field_value": start_date,
+            },
         ]
-        params.extend(self.generate_params(conditions))
+
+        query_builder = (
+            TimeEntryQueryBuilder()
+            .add_sql_interception_with_date_range_condition(
+                start_date, end_date
+            )
+            .add_sql_where_not_equal_condition(not_equal_conditions)
+            .add_sql_where_equal_condition(conditions)
+            .add_sql_ignore_id_condition(ignore_id)
```

Comment:
This ignore id condition seems to make sens only in time-entries, if so it could be implemented in timeentries query builder directly. Also please if some top-level function is actually using this condition, if not you can remove it.

Reply:
Resolved with the changes agreed at the meeting.